packages/vaadin-icon/src/vaadin-icon.js

```diff
@@ -138,17 +138,48 @@ class IconElement extends ThemableMixin(ElementMixin(PolymerElement)) {
     return ['__svgChanged(svg, __svgElement)'];
   }
 
+  /** @private */
+  __getIconSetName(icon) {
+    if (!icon) {
+      return;
+    }
+
+    const parts = icon.split(':');
+    return parts[0] || DEFAULT_ICONSET;
+  }
+
+  constructor() {
+    super();
+    this.__iconsetDefinedListener = this.__iconsetDefinedListener.bind(this);
+  }
+
   /** @protected */
   ready() {
     super.ready();
     this.__svgElement = this.shadowRoot.querySelector('svg');
   }
 
+  /** @private */
+  __iconsetDefinedListener(e) {
+    if (e.detail === this.__getIconSetName(this.icon)) {
+      this.__iconChanged(this.icon);
+    }
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    document.addEventListener('vaadin-iconset-defined', this.__iconsetDefinedListener);
```

Comment:
_In a perfect world, the event listener could be added only if the icon set was not defined, but I don't see any problems with the current solution as well. So let's not dwell on it :)_

Reply:
note that the `icon` property can be changed to use another iconset dynamically, one that may not yet be defined